### PR TITLE
Adds functionality for separating files based on the state of their c…

### PR DESCRIPTION
**Pull request by RichardYSun**
…hanges

  (ie: To Redact, In Progress, In Review, Complete)
  New docs are all in To Redact
  Any changes to redactions sets doc to In Progress
  Once user clicks "Mark As Complete" button in In Progress state, doc set to In Review (I don't think this should be the case, but we can talk about the logic here)
  Once user clicks "Mark As Complete" button in In Review state, doc set to Completed

https://github.com/wendi-yu/newbloom/assets/46700318/74c8cba7-de69-4838-ac81-86645e48954a

### CODE REVIEW

File: frontend/src/pages/Home.jsx [hunk 1]
```diff
@@ -160,21 +161,26 @@ const Home = () => {
   });
 
   // state management for which layout to display
-  const [docLayout, setDocLayout] = useState(HOMEPAGE_DOC_LAYOUTS.grid);
+  const [docLayout, setDocLayout] = useState(HOMEPAGE_DOC_LAYOUTS.grid)
+  const [docSection, updateDocSection] = useState(LocalStore.DOC_STATE.New)
+
+  var docsInSection = docState.docs.filter(doc => {
```
Comment (wendi-yu) [inline frontend/src/pages/Home.jsx:167]: `let`, not `var`

File: frontend/src/pages/Home.jsx [hunk 2]
```diff
@@ -160,21 +161,26 @@ const Home = () => {
...
+    return doc.docState == docSection
```
Comment (wendi-yu) [inline frontend/src/pages/Home.jsx:168]: `===`, not `==`

nit but you don't really need the `return` here, you can go
`.filter(doc => doc.docState === docSection)`

File: frontend/src/util/localDocStore.js [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+
 const addDocument = (name, id, state) => {
   const userId = getCurrentUser();
   // todo: this only allows one path per document
 
   let docHashesString = localStorage.getItem(userId);
   const docHashes = docHashesString ? JSON.parse(docHashesString) : {};
   // todo: warning if we''re going to overwrite
-  docHashes[id] = { name: name, state: state, dateAdded: new Date(), comments:[] };
+  docHashes[id] = { name: name, state: state, docState: DOC_STATE.New, dateAdded: new Date() };
```
Comment (wendi-yu) [inline frontend/src/util/localDocStore.js:17]: put the comments back, it'll break the rest of our code

File: frontend/src/util/localDocStore.js [hunk 4]
```diff
@@ -23,6 +30,7 @@ const updateDocumentBody = (docId, state) => {
 
   const docHashes = JSON.parse(docHashesString);
   docHashes[docId].state = state;
+  docHashes[docId].docState = DOC_STATE.InProgress;
```
Comment (wendi-yu) [inline frontend/src/util/localDocStore.js:33]: only if it's not completed? unless we want edits on completed docs to put them back into in progress

File: frontend/src/util/localDocStore.js [hunk 5]
```diff
@@ -108,6 +116,7 @@ export const getLocalDocuments = () => {
       owner: userId,
       documentBody: obj.state,
       comments: obj.comments || [],
+      docState: obj.docState ? obj.docState : DOC_STATE.New,
```
Comment (wendi-yu) [inline frontend/src/util/localDocStore.js:119]: very much a nit but maybe we want to call it `status` and not `docState` bc that's easy to mix up with `documentBody`

**Review by wendi-yu — APPROVED** [review]

File: frontend/src/util/localDocStore.js (hunk 4 above)
Comment (RichardYSun) [reply frontend/src/util/localDocStore.js:33]: I mean, that's what how I thought about it. Like, sure :)

**Pull request merged**